### PR TITLE
docs: Add documentation for Bun.sh

### PR DESCRIPTION
**Pull request by Pietrucci-Blacher**
## Description
Hi again, here's an other proposal to add documentation for Bun.sh js runtime.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -143,6 +157,14 @@ For example, you can do the following to have Prettier run before each commit:
    node --eval "fs.writeFileSync('.husky/pre-commit','pnpm exec lint-staged\n')"
    ```
 
+  <!--bun-->
+
+   ```bash
+   bun add --dev husky lint-staged
+   bunx husky init
+   bun --eval "fs.writeFileSync('.husky/pre-commit','pnpm exec lint-staged\n')"
```
Comment (fisker) [inline docs/install.md:165]:
```suggestion
   bun --eval "fs.writeFileSync('.husky/pre-commit','bunx lint-staged\n')"
```

File: docs/install.md [hunk 2]
```diff
@@ -143,6 +157,14 @@ For example, you can do the following to have Prettier run before each commit:
    node --eval "fs.writeFileSync('.husky/pre-commit','pnpm exec lint-staged\n')"
    ```
 
+  <!--bun-->
+
+   ```bash
+   bun add --dev husky lint-staged
+   bunx husky init
```
Comment (fisker) [inline docs/install.md:164]: As I tested on Windows this command fails 

```js
>bunx husky init
<CWD>/node_modules/husky/bin.mjs:18
        w('.husky/pre-commit', p.env.npm_config_user_agent.split('/')[0] + ' test\n')
                                                           ^

TypeError: Cannot read properties of undefined (reading 'split')
    at <CWD>/node_modules/husky/bin.mjs:18:53
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)

Node.js v20.10.0
```

Node.js issue?

File: docs/install.md (hunk 1 above)
Comment (Pietrucci-Blacher) [reply docs/install.md:165]: Ah, my bad, i'm going to change it

**Review by fisker — APPROVED** [review]

**Comment by Pietrucci-Blacher** [comment]
Hello Fisker, sorry to haven't fixed it yet, quite occupated, thanks for fixing this one 👌

File: docs/install.md (hunk 2 above)
Comment (Pietrucci-Blacher) [reply docs/install.md:164]: I've also managed to reproduce this issue on the latest version of Node.js, so I'm also wondering about it.

**Pull request merged**